packages/machine/test/integration/setup-then-set-state.spec.ts

```diff
@@ -0,0 +1,204 @@
+import AppRegistry from "@counterfactual/contracts/build/contracts/AppRegistry.json";
+import ETHBucket from "@counterfactual/contracts/build/contracts/ETHBucket.json";
+import MinimumViableMultisig from "@counterfactual/contracts/build/contracts/MinimumViableMultisig.json";
+import NonceRegistry from "@counterfactual/contracts/build/contracts/NonceRegistry.json";
+import ProxyFactory from "@counterfactual/contracts/build/contracts/ProxyFactory.json";
+import StateChannelTransaction from "@counterfactual/contracts/build/contracts/StateChannelTransaction.json";
+
+import { AssetType, NetworkContext } from "@counterfactual/types";
+import { Contract, Wallet } from "ethers";
+import { AddressZero, WeiPerEther, Zero } from "ethers/constants";
+import { JsonRpcProvider } from "ethers/providers";
+import {
+  BigNumber,
+  hexlify,
+  Interface,
+  randomBytes,
+  SigningKey
+} from "ethers/utils";
+
+import {
+  SetStateCommitment,
+  SetupCommitment
+} from "../../src/middleware/protocol-operation";
+import { StateChannel } from "../../src/models";
+
+// To be honest, 30000 is an arbitrary large number that has never failed
+// to reach the done() call in the test case, not intelligency chosen
+const JEST_TEST_WAIT_TIME = 30000;
+
+// ProxyFactory.createProxy uses assembly `call` so we can't estimate
+// gas needed, so we hard-code this number to ensure the tx completes
+const CREATE_PROXY_AND_SETUP_GAS = 6e9;
+
+// Similarly, the SetupCommitment is a `delegatecall`, so we estimate
+const SETUP_COMMITMENT_GAS = 6e9;
+
+// The AppRegistry.setState call _could_ be estimated but we haven't
+// written this test to do that yet
+const SETSTATE_COMMITMENT_GAS = 6e9;
+
+let networkId: number;
+let provider: JsonRpcProvider;
+let wallet: Wallet;
+
+// TODO: When we add a second use case of this custom mathcer,
+//       move it and its typigns into somewhere re-usable
+declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toBeEq(expected: BigNumber): BigNumber;
+    }
+  }
+}
+expect.extend({
+  toBeEq(received: BigNumber, argument: BigNumber) {
+    return {
+      pass: received.eq(argument),
+      message: () => `expected ${received} not to be equal to ${argument}`
+    };
+  }
+});
+
+// TODO: This will be re-used for all integration tests, so
+//       move it somewhere re-usable when we add a new test
+beforeAll(async () => {
+  require("dotenv-safe").config();
```

Comment:
The `require` is inconsistent with how other dependencies are used

Reply:
I've done it this way to mimic the way it is most often used in example code, docs, other codebases etc. We could make it conform to the `import` style but somehow I feel this is more expressive and easy to read? What do you think

Reply:
```suggestion
import("dotenv-safe").config()
```

Does the same thing and retains consistent style. This is not just a code aesthetics thing, the entire codebase uses ES6 import/exports so doing a CommonJS require feels out of place.